### ISSUE TITLE
feature request: canDeselect promise for validation purposes 

### ISSUE THREAD

**Issue by entrodus** [valor-software/ng2-tree#155]
It would be very handy to have a method for controlling whether a currently selected node can be deselected through a promise/observable function which could be used for validation purposes. 

I am currently using ng2-tree component to visualise all of my entities on the left side and display all editable properties of the selected entity on the right side of my application. 

It would be great if i could hook to a canDeselect promise before user navigates to a new node so i could apply validation checks and user prompts for saving currently selected node before navigating to a new one.

**Comment by rychkog** [comment]
@entrodus So, you need smth like a guard, that checks asynchronously whether it is allowed to deselect a node, right?

**Comment by entrodus** [comment]
@rychkog Yes. A guard that would also not allow the selection of a new node until the deselection promise of the previous node is resolved and satisfied.
